app/views/users/edit.html.erb

```diff
@@ -31,3 +31,7 @@
                email_addresses: current_user.email_addresses } %>
   </div>
 </div>
+
+<% if controller.request.env["HTTP_REFERER"].present? %>
+  <%= link_to 'Back', :back %>
```

Comment:
Can it be an explicit link to that referrer?

@Dantemss you are the king of all things having to do with passing return URLs around -- what do you think?